manifest/list.go

```diff
@@ -0,0 +1,119 @@
+package manifest
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/containers/image/types"
+	digest "github.com/opencontainers/go-digest"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// ManifestList is an interface for parsing, modifying lists of image manifests.
+// Callers can either use this abstract interface without understanding the details of the formats,
+// or instantiate a specific implementation (e.g. manifest.OCI1Index) and access the public members
+// directly.
+type ManifestList interface {
+	// MIMEType returns the MIME type of this particular manifest list.
+	MIMEType() string
+
+	// Instances returns a list of the manifests that this list knows of.
+	Instances() []types.BlobInfo
+
+	// Update information about the list's instances.  The length of the passed-in slice must
+	// match the length of the list of instances which the list already contains, and every field
+	// must be specified.
+	UpdateInstances([]ManifestListUpdate) error
+
+	// ChooseDigest selects which manifest is most appropriate for the platform described by the
+	// SystemContext, or for the current platform if the SystemContext doesn't specify any details.
+	ChooseDigest(ctx *types.SystemContext) (digest.Digest, error)
+
+	// ImageID computes a recommended image ID based on the list of images referred to by the manifest.
+	ImageID() string
+
+	// Serialize returns the list in a blob format.
+	// NOTE: Serialize() does not in general reproduce the original blob if this object was loaded
+	// from, even if no modifications were made!
+	Serialize() ([]byte, error)
+
+	// ToOCI1Index() returns the list rebuilt as an OCI1 index, converting it if necessary.
+	ToOCI1Index() (*OCI1Index, error)
+
+	// ToSchema2List() returns the list rebuilt as a Schema2 list, converting it if necessary.
+	ToSchema2List() (*Schema2List, error)
```

Comment:
Simple enough.  Added it.

Reply:
Could `copyMultipleImages` call `ConvertToMIMEType` now?

Reply:
Could this go one step further, then, and having only the `ConvertToMIMEType` method public, and having each individual implementation its own dispatch into private conversion implementations, without everything indirecting through `manifest.ConvertListToMIMEType?

Having one `switch` per implementation should not be that much more code, and it would allow keeping the `List` interface type-agnostic (apart from hard-coding the concept of a manifest list and assumptions about what that means, sure).

Reply:
Looks reasonable.  Changing it.